src/main/java/uk/ac/ebi/eva/pipeline/io/writers/AnnotationMongoWriter.java

```diff
@@ -0,0 +1,146 @@
+import org.springframework.data.mongodb.core.query.BasicUpdate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.util.Assert;
+import uk.ac.ebi.eva.commons.models.converters.data.AnnotationToSimplifiedDBObjectConverter;
+import uk.ac.ebi.eva.commons.models.mongo.documents.Annotation;
+import uk.ac.ebi.eva.commons.models.mongo.documents.subdocuments.ConsequenceType;
+import uk.ac.ebi.eva.commons.models.mongo.documents.subdocuments.Xref;
+import uk.ac.ebi.eva.utils.MongoDBHelper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static uk.ac.ebi.eva.commons.models.mongo.documents.Annotation.CONSEQUENCE_TYPE_FIELD;
+import static uk.ac.ebi.eva.commons.models.mongo.documents.Annotation.XREFS_FIELD;
+
+/**
+ * Write a list of {@link Annotation} into MongoDB
+ * <p>
+ * A new annotation is added in the existing document.
+ * In case of two annotations (or more) in the same variant the other annotations are appended:
+ * <p>
+ * 20_63963_G/A    20:63963   A  ENSG00000178591    ENST00000382410    Transcript upstream_gene_variant  -  -  -  -  -  -  DISTANCE=4388;STRAND=1;SYMBOL=DEFB125;SYMBOL_SOURCE=HGNC;HGNC_ID=18105;BIOTYPE=protein_coding;CANONICAL=YES;CCDS=CCDS12989.2;ENSP=ENSP00000371847;SWISSPROT=DB125_HUMAN;TREMBL=B2R4E8_HUMAN;UNIPARC=UPI00001A36DE
+ * 20_63963_G/A    20:63963   A  ENSG00000178591    ENST00000608838    Transcript upstream_gene_variant  -  -  -  -  -  -  DISTANCE=3928;STRAND=1;SYMBOL=DEFB125;SYMBOL_SOURCE=HGNC;HGNC_ID=18105;BIOTYPE=processed_transcript
+ * <p>
+ * Will be:
+ * <p>
+ * "annot" : {
+ * "ct" : [
+ * { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000382410", "codon" : "-", "strand" : "+", "bt" : "protein_coding", "aaChange" : "-", "so" : [ 1631 ] },
+ * { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000608838", "codon" : "-", "strand" : "+", "bt" : "processed_transcript", "aaChange" : "-",
+ * "so" : [ 1631 ] } ],
+ * "xrefs" : [
+ * { "id" : "DEFB125", "src" : "HGNC" },
+ * { "id" : "ENST00000382410", "src" : "ensemblTranscript" },
+ * { "id" : "ENST00000608838", "src" : "ensemblTranscript" },
+ * { "id" : "ENSG00000178591", "src" : "ensemblGene"
+ */
+public class AnnotationMongoWriter implements ItemWriter<Annotation> {
+
+    private static final String ANNOTATION_XREF_ID_FIELD = Annotation.XREFS_FIELD + "." + Xref.XREF_ID_FIELD;
+
+    private static final String ANNOTATION_CT_SO_FIELD = Annotation.CONSEQUENCE_TYPE_FIELD + "."
+            + ConsequenceType.SO_ACCESSION_FIELD;
+    public static final String EACH = "$each";
+    public static final String ADD_TO_SET = "$addToSet";
+
+    private final MongoOperations mongoOperations;
+
+    private final String collection;
+
+    private final AnnotationToSimplifiedDBObjectConverter converter;
+
+    public AnnotationMongoWriter(MongoOperations mongoOperations, String collection) {
+        super();
```

Comment:
Nah, it's call let's remove this extension of the base class and forget to remove the thisis in the constructor.